tagged.cabal

```diff
@@ -55,6 +55,9 @@ library
   hs-source-dirs: src
   exposed-modules: Data.Tagged
 
+  if impl(ghc >= 8.10)
+    ghc-options: -Winferred-safe-imports -Wmissing-safe-haskell-mode
```

Comment:
What do these flags do? I don't see them referenced in the [User's Guide section on warnings](https://downloads.haskell.org/ghc/8.10.3/docs/html/users_guide/flags.html#warnings).

Reply:
You are looking in the wrong section :) https://downloads.haskell.org/ghc/8.10.3/docs/html/users_guide/safe_haskell.html#ghc-flag--Winferred-safe-imports

Reply:
Hmm.. https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3295, dunno if you want to remove it for now to be on the safer side.

Reply:
I still find it rather bizarre that these flags don't show up in the flag reference, especially since that is the first place I look to find a flag that I'm unfamiliar with. I've opened [GHC#19132](https://gitlab.haskell.org/ghc/ghc/-/issues/19132) for this.

In any case, I'll reiterate usual mantra about Safe Haskell from [here](https://github.com/ekmett/semigroupoids/issues/69#issuecomment-361304259): I'm fine with accepting PRs that work towards improving Safe Haskell support, but if things break in the future, it'll be up to you to repair the damage.